query/src/test/scala/filodb/query/exec/WindowIteratorSpec.scala

```diff
@@ -330,4 +324,138 @@ class WindowIteratorSpec extends RawDataWindowingSpec {
         .asInstanceOf[ChunkedRangeFunction], queryConfig)()
     chunkedWinIt.map(r => (r.getLong(0), r.getDouble(1))).toList.filter(!_._2.isNaN) shouldEqual windowResults
   }
+
+  it("should calculate AvgOverTime correctly") {
```

Comment:
how about "correctly even for windows with no values or NaN only"?  just a bit more info to make case precise

Reply:
Done